libr/anal/sign.c

```diff
@@ -837,6 +837,229 @@ R_API bool r_sign_delete(RAnal *a, const char *name) {
 	return sdb_remove (a->sdb_zigns, k, 0);
 }
 
+static bool matchBytes(RSignItem *a, RSignItem *b) {
+	if (a->bytes && b->bytes) {
+		if (a->bytes->size == b->bytes->size) {
+			return !memcmp (a->bytes->bytes, b->bytes->bytes, b->bytes->size);
+		}
+	}
+	return false;
+}
+
+static bool matchGraph(RSignItem *a, RSignItem *b) {
+	if (a->graph && b->graph) {
+		if (a->graph->cc != b->graph->cc) {
```

Comment:
i would have a counter incrementing everytime there's a match ini here, at the end you can check the % of matching ratio to return true or false